src/background/service/keyring/eth-keystone-keyring.ts

```diff
@@ -48,12 +48,11 @@ export default class KeystoneKeyring extends MetaMaskKeyring {
     super();
 
     this.getInteraction().on(AcquireMemeStoreData, () => {
-      this.getMemStore().subscribe((data) => {
-        const request = data.sign?.request;
-        if (request) {
-          this.getInteraction().emit(MemStoreDataReady, request);
-        }
-      });
+      const data = this.getMemStore().getState();
```

Comment:
This commit doesn't work for me. Is that working for you?

Reply:
No, I did some more testing and ran into some issues that I'm trying to work out. I think I don't quite understand the event interactions well enough.

Reply:
https://github.com/RabbyHub/Rabby/pull/2202  This fix doesn't work for you?

Reply:
No it doesn't. That change fixes it so that the first signing attempt is valid, but afterwards it starts sending old transactions to the wallet.
I pushed some more changes to this branch that seem to resolve the issue (without the changes from #2022 even).